### PR TITLE
Do not collect symlinked tests under Windows

### LINKED ISSUE #12039: Pytest 8.0.2 may collect tests multiple times under Windows
I'm not quite sure if this is a bug, or a problem with the test or GitHub Actions, but since version 8.0.2 a test in `pytest-order` fails in the CI under Windows, which worked fine in 8.0.1 and previous versions. Linux and macOS tests are not affected.

Pytest run environment in the CI:
```
platform win32 -- Python 3.11.8, pytest-8.0.2, pluggy-1.4.0 -- D:\a\pytest-order\pytest-order\.tox\py311-pytest80\Scripts\python.EXE
cachedir: .tox\py311-pytest80\.pytest_cache
rootdir: D:\a\pytest-order\pytest-order
plugins: cov-2.9.0, dependency-0.6.0, mock-3.12.0, xdist-3.5.0
```
The same happens for Python 3.8 - 3.12.

The [actual test](https://github.com/pytest-dev/pytest-order/blob/main/tests/test_xdist_handling.py) should collect 6 items (3 tests per test module, in 2 modules) that are created in the directory provided by the `tempdir` fixture, 
but collects 24 tests in the CI instead. Locally, it works as expected.

Checking the failed tests revealed that the tests have been collected from different locations. The tempdir had been at:
`C:\\Users\\runneradmin\\AppData\\Local\\Temp\\pytest-of-runneradmin\\pytest-9\\test_xdist_ordering0` 
(for a specific run), but the same tests have been discovered with a test ID with just the qualified test name, and additionally with the test IDs:
`::::Users::runneradmin::AppData::Local::Temp::pytest-of-runneradmin::pytest-9:::xxx` and 
`::::Documents and Settings::runneradmin::AppData::Local::Temp::pytest-of-runneradmin::pytest-9::xxx`,
which are most likely aliases.

I haven't dug into this yet, but I suspect it may have something to do with the fix for [the problem with short paths](https://github.com/pytest-dev/pytest/issues/11895).

### PR DESCRIPTION
**Pull request by mrbean-bremen**
The check for short paths under Windows via os.path.samefile, introduced in #11936, also found similar tests in symlinked tests in the GH Actions CI. This checks additionally that one of the files is not a symlink.

Fixes #12039.

### CODE REVIEW

File: testing/test_collection.py [hunk 1]
```diff
@@ -1789,6 +1789,26 @@ def test_collect_short_file_windows(pytester: Pytester) -> None:
     assert result.parseoutcomes() == {"passed": 1}
 
 
+def test_not_collect_symlink_syblings(
+    pytester: Pytester, tmp_path: Path, request: pytest.FixtureRequest
```
Comment (mrbean-bremen) [inline testing/test_collection.py]: Thanks @nicoddemus, that's much better! Nitpicking: its "siblings", not "syblings".

**Comment by nicoddemus** [comment]
Thanks @mrbean-bremen!

Tweaked things a bit to add more information (I also added a test similar to yours in a previous commit, because I saw the PR yesterday without a test, and went to work on a test right away this morning, not realizing you had pushed a new test since then 😅).

File: testing/test_collection.py (hunk 1 above)
Comment (nicoddemus) [reply testing/test_collection.py]: Thanks!

**Comment by mrbean-bremen** [comment]
Yes, I added the test before making the PR - you have probably seen the branch before that. Anyway, your test is better, and the documentation is clearer :)

**Review by nicoddemus — APPROVED** [review]
Thanks again @mrbean-bremen!

**Pull request merged**